logictestccl: avoid static identifiers for enums

otan opened the pull request:
We cannot rely on an enum having a static ID in logic tests. As such,
rework tests that rely on having a static ID.

Resolves #67869

Release justification: test only change

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/69225)
<!-- Reviewable:end -->

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/69225#-:-MhkacrMCf3IPu7ZJmb_:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajstorm and @arulajmani)



<!-- Sent from Reviewable.io -->

otan commented:
thanks!

bors r=ajwerner

otan commented:
bors r-

release justifications begin...

otan commented:
bors r=ajwerner

otan committed:
logictestccl: avoid static identifiers for enums

We cannot rely on an enum having a static ID in logic tests. As such,
rework tests that rely on having a static ID.

Release justification: test only change

Release note: None

otan commented:
bors r=ajwerner

otan commented:
bors retry
